Add demo example in the quickstart guide (#355)
containous/maesh: docs/content/quickstart.md

Assuming `tiller` is deployed in your `kube-system` namespace, this will
give it very open permissions.

## Installing Maesh

> ```bash tab="Command"

Reviewer: If there's a command tab, I guess there should be an expected output as well?

Author: Way too long, isn't it ?

Reviewer 2: You could also do a `[...]` truncated output, just to show the top and bottom perhaps...

Author: Yep okay.

helm repo add maesh https://containous.github.io/maesh/charts
helm repo update
helm install --name=maesh --namespace=maesh maesh/maesh
```

## Using Maesh

> Let's deploy a server application and a client application under the `maesh-test` namespace.

Reviewer: Maybe we should make it more obvious that this is an example and not the way they have to use Maesh 😅

```suggestion
As an example, let's deploy a server application and a client application under the `maesh-test` namespace.
```

service/server   ClusterIP   10.43.17.247   <none>        80/TCP    77s

NAME                     READY     UP-TO-DATE   AVAILABLE   AGE
deployment.apps/client   1/1       1            1           79s
deployment.apps/server   2/2       2            2           77s

NAME                                DESIRED   CURRENT   READY     AGE
replicaset.apps/client-7446fdf848   1         1         1         79s
replicaset.apps/server-7c8fd58db5   2         2         2         77s
```

Take note of the client app pod name (here it's `client-7446fdf848-x96fq`) and open a new terminal session inside this pod using `kubectl exec`.

```bash
kubectl -n maesh-test exec -ti client-7446fdf848-x96fq ash
```

From inside the client container make sure you are able to reach your server using kubernetes DNS service discovery.

```bash tab="Command"
curl server.maesh-test.svc.cluster.local
```

```test tab="Expected Output"
Hostname: server-7c8fd58db5-sd4f9
IP: 127.0.0.1
IP: ::1
IP: 10.42.2.10
IP: fe80::a4ec:77ff:fe37:1cdd
RemoteAddr: 10.42.2.9:46078
GET / HTTP/1.1
Host: server.maesh-test.svc.cluster.local
User-Agent: curl/7.64.0
Accept: */*
```

> You can note that all this server application is doing is to answer back the content of the request it receives.

Reviewer: ```suggestion
You can note that all this server application is doing is to respond with the content of the request it receives.
```

Now replace the `svc.cluster.local` suffix by `maesh`, and tada: you are now using Maesh to reach your server!

```bash tab="Command"
curl server.maesh-test.maesh
```

```test tab="Expected Output"
Hostname: server-7c8fd58db5-rchg8
IP: 127.0.0.1
IP: ::1
IP: 10.42.1.7
IP: fe80::601d:7cff:fe26:c8c6
RemoteAddr: 10.42.1.5:59478
GET / HTTP/1.1
Host: server.maesh-test.maesh
User-Agent: curl/7.64.0
Accept: */*
Accept-Encoding: gzip
Uber-Trace-Id: 3f9e7129a059f70:7e889a1ebcb147ac:3f9e7129a059f70:1
X-Forwarded-For: 10.42.2.9
X-Forwarded-Host: server.maesh-test.maesh
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: maesh-mesh-w95q2
X-Real-Ip: 10.42.2.9
```

> Note the presence of `X-Forwarded` headers as well than other instrumentation headers like `Uber-Trace-Id`, indicating than your request has been processed and instrumented by Maesh.

Reviewer: ```suggestion
Note the presence of `X-Forwarded` headers as well as other instrumentation headers like `Uber-Trace-Id`, indicating than your request has been processed and instrumented by Maesh.
```
